project-management/src/main/java/life/qbic/projectmanagement/application/ProjectInformationService.java

```diff
@@ -91,6 +92,12 @@ public Optional<Project> find(String projectId) throws IllegalArgumentException
     return find(ProjectId.parse(projectId));
   }
 
+  public boolean isProjectCodeUnique(String projectCode) throws IllegalArgumentException {
+    return projectRepository.find(ProjectCode.parse(projectCode)).stream()
+        .findFirst()
+        .isEmpty();
```

Comment:
Why a list and not an optional?

Reply:
Good point, i used the current implementation which was list based. I replaced it with optional which makes more sense since projectcodes are supposed to be unique 🤔